app/models/participant.rb

```diff
@@ -128,7 +128,7 @@ def update_faye
   end
 
   def has_new_visit_groups?
-    self.arm.visit_groups.count > self.appointments.where(arm_id: self.arm_id).count
+    self.arm.visit_groups.map(&:id).sort != self.appointments.where(arm_id: self.arm_id).where.not(visit_group_id: nil).map(&:visit_group_id).sort
```

Comment:
self.arm.visit_groups.ids.order('id') != self.appointments.where(arm_id: self.arm_id).where.not(visit_group_id: nil).visit_group_ids.order('id')